main.py

```diff
@@ -1,10 +1,65 @@
+from flask import request
+from flask import Flask, jsonify
+
 from database_connector import DatabaseConnector
 from database_manager import ItemManager, TagManager
+from search_manager import SearchManager
+from product_manager import ProductManager
 
 database_connector = DatabaseConnector('db')
 connection = database_connector.get_connection()
 item_manager = ItemManager(database_connector)
 tag_manager = TagManager(database_connector)
+search_manager = SearchManager(tag_manager, item_manager)
+product_manager = ProductManager(item_manager, tag_manager)
+
+app = Flask(__name__)
+
+
+@app.route('/')
+def index():
+    return 'Hello, World, this is the index!'
+
+
+@app.route('/unique_tags', methods=['GET', 'POST'])
+def unique_tags():
+    if request.method == 'GET':
+        return jsonify(tag_manager.get_unique_tags_list())
+    else:
+        return "no tags requested"
+
+
+@app.route('/add_product', methods=['GET', 'POST'])
+def add_product():
+    if request.method == 'GET':
+        # TODO: updated tags below to get data from FE
+        tags = ["bird", "hello"]
+        product_manager.add_product(tags)
+        return 'A product has been added'
+    else:
+        return "no button clicked yet to add a product"
+
+
+@app.route('/delete_product', methods=['GET', 'POST'])
+def delete_product():
+    if request.method == 'GET':
+        # TODO: update POST and to recieve item_id from FE
+        item_id = "6baa81a4-5a9d-4130-bc5f-1e842d1ecb53"
+        product_manager.delete_product(item_id)
+        return "A product has been deleted"
+    else:
+        return "no button clicked yet to delete a product"
+
+
+@app.route('/search', methods=['GET', 'POST'])
+def search():
+    # TODO: hook up actual request argument below ones FE is a built and update to POST
+    search_word = "sw"
+    if request.method == 'GET':
+        return jsonify(search_manager.get_all_maching_products(search_word))
+    else:
+        return "no word submitted yet"
```

Comment:
You should always return the same type, not sometimes javascript/json, and sometimes text/html. Perhaps return a dict that has two values, one with the status/success info, and one with the data.

Reply:
removed the else as it wasn't needed. But let's discuss where I should implement some fail/error messages for my database queries as I've yet to do that